src/code/z_sub_s.c

```diff
@@ -68,7 +135,22 @@
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_sub_s/func_8013D924.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sub_s/func_ActorCategoryIterateById.s")
+/**
+ * Finds the first actor instance of a specified Id and category.
+ */
+Actor* SubS_ActorFind(GlobalContext* globalCtx, Actor* actorListStart, u8 actorCategory, s16 actorId) {
```

Comment:
Have you considered swapping the order of the names of these functions? Something like:
`SubS_FindActor`, `SubS_FindNearestActor`, `SubS_FindActorCustom`, and `SubS_FindDoor`? Just a thought, either is fine with me

Reply:
Those do seem better so going to go with those.